### PR TITLE
[cpplint] Enable new recommended configuration

### LINKED ISSUE #1584: [cpplint] Enable new recommended configuration
Related to https://github.com/sider/sideci/issues/8876

Now that the notification period expires, we will enable the recommended configuration.

https://github.com/sider/runners/blob/2ed2c84a160ced4002b19feb73102b857f53fab8/lib/runners/processor/cpplint.rb#L28-L42

### PR DESCRIPTION
**Pull request by suetsuguk**
Fix #1584

> Check the following items.

- [x] Add a new [changelog](https://github.com/sider/runners/blob/master/CHANGELOG.md) entry if this change is notable.

If you are adding a new runner:

- ~~[x] Read and follow the [document](https://github.com/sider/runners/blob/master/docs/how-to-write-a-new-runner.md) for a new runner.~~
- ~~[x] Add a new tool to [`analyzers.yml`](https://github.com/sider/runners/blob/master/analyzers.yml).~~
- ~~[x] Run `bundle exec rake readme:generate` and commit the updated `README.md`.~~
- ~~[x] Add a new tool to the [Dependabot configuration](https://github.com/sider/runners/blob/master/.github/dependabot.yml) if needed.~~
- ~~[x] Provide option(s) that users can customize.~~
- [x] Write smoke test cases. E.g. all options, successful end, warnings, errors, etc.
- ~~[x] Add a [CI setting](https://github.com/sider/runners/blob/master/.github/workflows/build.yml).~~
- ~~[x] Write a [document](https://github.com/sider/sider-docs) and link the pull request, e.g. `sider/sider-docs#789`~~

### CODE REVIEW

**Review by yykamei — APPROVED** [review]
LGTM

Could you add an explanation for the changes of `sider_recommended_CPPLINT.cfg` from #1505?

File: lib/runners/processor/cpplint.rb [hunk 1]
```diff
@@ -26,17 +26,11 @@ class Processor::Cpplint < Processor
     CONFIG_FILE_NAME = "CPPLINT.cfg".freeze
 
     def setup
-      # TODO: When the notification period expires, comment out the below line (warn_recommended_config_file_release)
-      #       and uncomment the second line (deploy_recommended_config_file) to activate our default configuration file.
-      warn_recommended_config_file_release(CONFIG_FILE_NAME, "in mid October 2020")
-
-=begin
       if config_linter[:filter]
         trace_writer.message "The `filter` option in #{config.path_name} is specified. The Sider's recommended ruleset is ignored."
```
Comment (ybiquitous) [inline lib/runners/processor/cpplint.rb:30]: [nitpick]

```suggestion
        trace_writer.message "The `filter` option in `#{config.path_name}` is specified. The Sider's recommended ruleset is ignored."
```

**Review by ybiquitous — APPROVED** [review]
LGTM

**Comment by suetsuguk** [comment]
> Could you add an explanation for the changes of `sider_recommended_CPPLINT.cfg` from #1505?

In the previous result, the analysis program treated the history of GitHub commits with wrong order.
After fixed the problem, we get proper and more "valid" results.
I have cross-checked the results with interview reviews.
For example, both `build/c++11` and `build/include_alpha`, which are newly added by the program, are marked as "un-useful" by two of four interviewees.
Other deletions such as `whitespace/ending_newline` and `whitespace/semicolon` are considered to be "useful" by the program. In fact, all four interviewees marked them as "useful".

**Comment by yykamei** [comment]
Thank you for your detailed explanation. The change is reasonable :+1:

File: lib/runners/processor/cpplint.rb (hunk 1 above)
Comment (suetsuguk) [reply lib/runners/processor/cpplint.rb]: I missed the quote. Thank you.
Fixed. commit f5b0e4380bc4057abc797025bd2e45e8918a786d

**Comment by suetsuguk** [comment]
Thank you all for taking your time. I will merge this PR into master.

**Pull request merged**